pkg/staticnetworkconfig/generator.go

```diff
@@ -340,8 +355,7 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistenceYAML(macIn
 			interfaceWithmacIdentifier[interfaceName.(string)] = struct{}{}
 		}
 	}
-
-	for _, iface := range Interfaces {
+	for _, iface := range interfacesSlice {
```

Comment:
not directly related to this PR, but why do we need to loop over the interfaces twice?

Reply:
Because of that:
`if _, exists := interfaceWithMacIdentifier[port.(string)]; exists { continue }`

When iterating over the interfaces, and there is a bond, we want to check if its ports are already listed in the YAML as interfaces with a MAC identifier. If they are, we don't need to include them in the MAC map, and we can skip them.

Reply:
Still I suppose we could simplify the logic with a single pass:/ But it's an unrelated refactor to this PR any way.